common/types.go

```diff
@@ -168,3 +169,71 @@ type LogConfig struct {
 	FileCount   uint64
 	TimeTracker bool
 }
+
+type Block struct {
+	Id         string
+	StartIndex int64
+	EndIndex   int64
+	Size       int64
+}
+
+// list that holds blocks containing ids and corresponding offsets
+type BlockOffsetList struct {
+	BlockOffsetList []*Block //blockId to offset mapping
```

Comment:
Name of the structure and this field are same is this intentional?

Reply:
is there a rule about this?

Reply:
there is no rule for that but structure and its member having same name does not make any sense logically.

Reply:
when you refer it it will be more list BlockOffsetList.BlockOffsetList[]